aiverify-foundation/moonshot-ui
# Fix edit endpoint form save button not enabled

## Description

In edit endpoint user flow, save button is not enabled when form values other than token, is updated. Issue caused by code changes in previous PR for endpoint form.
Code changes include correcting select input component label 'for' attribute and improvement to error util function.

## Motivation and Context

bug fix

## Type of Change



- fix: A bug fix









## How to Test

1. Go to endpoints screen
2. Select an endpoint and click 'edit endpoint' button
Save button should be disabled by default and enabled when form values are changed. Also verify endpoint is updated when save button is clicked.

Regression required for new endpoint flow.

## Checklist

Please check all the boxes that apply to this pull request using "x":

- [x ]  I have performed a self-review of my own code.
- [ x]  I have read, understood and agree to the Developer Certificate of Origin below, which this project utilises.

## Screenshots (if applicable)

[If the changes involve visual modifications, include screenshots or GIFs that demonstrate the changes.]

## Additional Notes

[Add any additional information or context that might be relevant to reviewers.]

<details>
  <summary>Developer Certificate of Origin</summary>

 ```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
 ```
</details>
